pkg/daemon/node.go

```diff
@@ -32,7 +32,13 @@ func (dn *Daemon) loadNodeAnnotations(node *corev1.Node) (*corev1.Node, error) {
 		currentOnDisk, err := dn.getCurrentConfigOnDisk()
 		if err == nil {
 			glog.Infof("Setting initial node config based on current configuration on disk: %s", currentOnDisk.GetName())
-			return setNodeAnnotations(dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, node.Name, map[string]string{constants.CurrentMachineConfigAnnotationKey: currentOnDisk.GetName()})
+			// this state shouldn't really happen, but since we're here, we're just going to set desired == current and state == done,
+			// otherwise the MCD will complain it cannot find those annotations
+			return setNodeAnnotations(dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, node.Name, map[string]string{
+				constants.CurrentMachineConfigAnnotationKey:     currentOnDisk.GetName(),
+				constants.DesiredMachineConfigAnnotationKey:     currentOnDisk.GetName(),
```

Comment:
Currently the node controller is the only thing setting the desired config.

I think this will introduce a race condition if a node completes an update and at the same time the node controller wants it to update to a different configuration.

The race might be benign - I think the node controller would probably just set it back.

But probably what we want is to only set it if it's not currently set.